### PR TITLE
git clone operation will be retried

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -876,3 +884,36 @@ def test_get_primary_images(tag_conf, tag_annotation, expected):
...
+            .should_receive('check_output')
+            .once()
+            .and_return(True))
+
+        (flexmock(subprocess)
+            .should_receive('check_call')
+            .once()
+            .and_raise(CustomTestException))
+
+    if raise_exc:
```
Comment (vrutkovs) [inline tests/test_util.py]:
```
exception = subprocess.CalledProcessError if raise_exc else CustomTestException
with pytest.raises(exception):
    clone_git_repo(DOCKERFILE_GIT, tmpdir_path, retry_times=retry_times)
```

File: atomic_reactor/util.py [hunk 2]
```diff
@@ -234,41 +235,40 @@ def wait_for_command(logs_generator):
...
+    cmd = ["git", "clone", git_url, quote(target_dir)]
+
+    for counter in range(retry_times + 1):
         try:
-            subprocess.check_call(cmd)
-        except subprocess.CalledProcessError as ex:
-            logger.warning(repr(ex))
-            # let's try again with plain `git clone $url && git checkout`
-            cmd = ["git", "clone", git_url, quote(target_dir)]
             logger.debug("cloning '%s'", cmd)
```
Comment (vrutkovs) [inline atomic_reactor/util.py]: Nitpick: move debug call out of the loop so it won't be repeated on retries

Comment (rcerven) [reply atomic_reactor/util.py]: ok

File: tests/test_util.py (hunk 1 above)
Comment (rcerven) [reply tests/test_util.py]: ok

File: atomic_reactor/util.py [hunk 3]
```diff
@@ -234,41 +235,40 @@ def wait_for_command(logs_generator):
...
-        except subprocess.CalledProcessError as ex:
-            logger.warning(repr(ex))
-            # let's try again with plain `git clone $url && git checkout`
-            cmd = ["git", "clone", git_url, quote(target_dir)]
-            logger.debug("cloning '%s'", cmd)
-            subprocess.check_call(cmd)
-            cmd = ["git", "reset", "--hard", commit]
-            logger.debug("checking out branch '%s'", cmd)
-            subprocess.check_call(cmd, cwd=target_dir)
+            subprocess.check_output(cmd, stderr=subprocess.STDOUT)
```
Comment (vrutkovs) [inline atomic_reactor/util.py]: `check_call` should be sufficient here - or, alternatively, log the returned output

Comment (rcerven) [reply atomic_reactor/util.py]: ok adding comment

**Comment by rcerven** [comment]
release note updated

**Pull request merged**